presto-main/src/main/java/com/facebook/presto/operator/aggregation/fixed_histogram/FixedDoubleBreakdownHistogram.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.fixedhistogram;
+
+import io.airlift.slice.SizeOf;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.operator.aggregation.fixedhistogram.FixedHistogramUtils.getIndexForValue;
+import static com.facebook.presto.operator.aggregation.fixedhistogram.FixedHistogramUtils.getLeftValueForIndex;
+import static com.facebook.presto.operator.aggregation.fixedhistogram.FixedHistogramUtils.getRightValueForIndex;
+import static com.facebook.presto.operator.aggregation.fixedhistogram.FixedHistogramUtils.verifyParams;
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * This is a fixed-bin histogram of weights breakdowns. For each bin, it stores the number of times
+ * each weight was seen.
+ *
+ * @apiNote This class is built for cases where the cardinality of the different weights is small, and
+ * the number of instances is very large. If the number of different weights added is not small, the time
+ * and space complexities can be very large.
+ */
+public class FixedDoubleBreakdownHistogram
+        implements Cloneable, Iterable<FixedDoubleBreakdownHistogram.BucketWeight>
```

Comment:
We don't need `Cloneable` I guess; same for all other classes

Reply:
Removed. Just out of curiosity, why not indicate that it implements some interfaces?

Reply:
That's a good point. The method `clone` itself is self-explanatory to indicate the caller to be aware of the data copy I guess? We can simplify the class structure given that.